CONTRIBUTING.md

```diff
@@ -11,56 +11,117 @@ Table of Contents:
+the general `rustfmt` rules can be followed for formatting.
 
-If any change has been made to the `config.rs` file, these changes should also be documented in the example configuration file `alacritty.yml`.
+If any change has been made to the `config.rs` file, these changes should also be documented in the
+example configuration file `alacritty.yml`.
 
-Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to things like code refactorings or documentation/tests) additionally need to be documented in the `CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be used to document changes from a user-perspective, instead of explaining the technical background (like commit messages). More information about Alacritty's change log format can be found [here](https://keepachangelog.com).
+Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to
+things like code refactorings or documentation/tests) additionally need to be documented in the
+`CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be
+used to document changes from a user-perspective, instead of explaining the technical background
+(like commit messages). More information about Alacritty's change log format can be found
+[here](https://keepachangelog.com).
 
 ### Style
 
-All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI build is failing because of formatting issues, you can install rustfmt using `rustup component add rustfmt` and then format all code using `cargo fmt`.
+All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI
+build is failing because of formatting issues, you can install rustfmt using `rustup component add
+rustfmt` and then format all code using `cargo fmt`.
+
+# Release Process
+
+Alacritty's release process aims to provide stable and well tested releases without having to hold
+back new features during the testing period.
+
+To achieve these goals, a new branch is created for every new release. Both the release candidates
+and the final version are only comitted and tagged in this branch. The master branch only tracks
+development versions, allowing us to keep the branches completely separate without merging releases
+back into master.
+
+The exact steps for an exemplary `1.2.3` might look like this:
+ 1. Initially, the version on the latest master is `1.2.3-dev`
+ 2. A new `v1.2.3` branch is created for the release
+ 3. On master, the version is bumped to `1.2.4-dev`
+        and the `-dev` is stripped from previous change log entries
+ 4. In the branch, the version is bumped to `1.2.3-rc1`
+ 5. The new commit in the branch is tagged as `1.2.3-rc1`
+ 6. A release is created for the `1.2.3-rc1` tag
+ 7. The changelog since the last release (stable or RC) is added
```

Comment:
I'm honestly not 100% sure what this means. I got to this step and had already updated the changelog's latest entry to `0.4.1-rc1` with all the latest changes from `0.4.1-dev`. This step seems like it might be a specific reminder to update the changelog on subsequent runs through the loop. If so, we should try and make this a bit more clear somehow... Maybe it could be as simple as "verify that the changelog include all cherry-picked fixes"? But then why not make this explicit in the next step where we talk about cherry-picking?

A cherry pick of a commit with a changelog entry should merge into the RC section no problem unless there's a case I'm missing. In my mock release this worked without any manual intervention.

Reply:
This is about the release's github description. I'll clarify.